rclcpp/src/rclcpp/node_interfaces/node_base.cpp

```diff
@@ -46,6 +48,9 @@ NodeBase::NodeBase(
   associated_with_executor_(false),
   notify_guard_condition_is_valid_(false)
 {
+  // Generate a mutex for this instance of NodeBase
+  this->map_object.create_mutex_of_nodebase(this);
```

Comment:
Is it standard practice to refer to static members using the instance reference like this? It seems misleading to me.

Reply:
Changed to use ```NodeBase::``` instead of ```this->```